state.go

```diff
@@ -1428,9 +1434,31 @@ func (e *UpdateRebootState) Handle(ctx *StateContext, c Controller) (State, bool
 
 	log.Debug("handling reboot state")
 
-	merr := c.ReportUpdateStatus(e.Update(), client.StatusRebooting)
-	if merr != nil && merr.IsFatal() {
-		return NewUpdateRollbackState(e.Update()), false
+	// Avoid attempting to report status to the server if any payload indicated a reboot of
+	// type `RebootTypeAutomaticSkipStatusReport` (even if other modules exist which
+	// use other reboot types, we need to assume that upstream connectivity will be broken by
+	// the payload which indicated `RebootTypeAutomaticSkipStatusReport`).
+	shouldPhoneHomeBeforeReboot := true
+	for n, _ := range c.GetInstallers() {
+		rebootRequested, err := e.Update().RebootRequested.Get(n)
+		if err != nil {
+			return e.HandleError(ctx, c, NewTransientError(errors.Wrap(
+				err, "Unable to get requested reboot type")))
+		}
+		if rebootRequested == datastore.RebootTypeAutomaticSkipStatusReport {
+			log.Warnf(
```

Comment:
nit: seems like this should be a debug

Reply:
Hm. For the moment this is custom Starry sauce so I thought it'd be nice to have this in the logs in the case that something does go wrong with this later. Maybe leave as-is until a time that upstream wants to accept this PR?